docs/content/docs/concepts/errors.mdx

```diff
@@ -0,0 +1,219 @@
+    } else if (isSolanaError(e, SOLANA_ERROR__TRANSACTION__EXCEEDS_SIZE_LIMIT)) {
+        console.error(
+            `Transaction exceeds size limit of ${e.context.transactionSizeLimit} bytes. ` +
+                `Actual size: ${e.context.transactionSize}`,
+        );
+    }
+    throw e;
+}
+```
+
+## Catching errors
+
+When you encounter an error, you can determine whether it is a `SolanaError` or not using the `isSolanaError()` function:
+
+```ts twoslash
+import { isSolanaError } from '@solana/kit';
+
+try {
+    // ...
+} catch (e) {
+    if (isSolanaError(e)) {
+        // A Solana error was thrown
+    } else {
+        // Something else went wrong
+    }
+}
+```
+
+When you have a strategy for handling a _particular_ error, you can supply that error's code as the second argument. This will both identify that error by code and refine the type of the `SolanaError` such that the shape of its `context` property becomes known to TypeScript.
+
+```ts twoslash
+import { SolanaError } from '@solana/kit';
+const e = null as unknown as SolanaError;
+// ---cut-before---
+import {
+    isSolanaError,
+    SolanaErrorCode,
+    SOLANA_ERROR__TRANSACTION__SIGNATURES_MISSING,
+} from '@solana/kit';
+
+try {
+    // ...
+} catch (e) {
+    if (isSolanaError(e, SOLANA_ERROR__TRANSACTION__SIGNATURES_MISSING)) {
+        //               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+        console.error(
+            // Now TypeScript knows the shape of this error's context
+            `Missing signatures for these addresses: [${e.context.addresses.join(', ')}]`,
+            //                                          ^^^^^^^^^
+        );
+    } else {
+        // Something else went wrong
+    }
+}
+```
+
+Some Solana errors have, as their root cause, another Solana error. One such example is during transaction simulation (preflight):
+
+```ts twoslash
+import { Base64EncodedWireTransaction, Rpc, SendTransactionApi } from '@solana/kit';
+const rpc = null as unknown as Rpc<SendTransactionApi>;
+const wireTransaction = null as unknown as Base64EncodedWireTransaction;
+// ---cut-before---
+import {
+    isSolanaError,
+    SOLANA_ERROR__INSTRUCTION_ERROR__COMPUTATIONAL_BUDGET_EXCEEDED,
+    SOLANA_ERROR__JSON_RPC__SERVER_ERROR_SEND_TRANSACTION_PREFLIGHT_FAILURE,
+} from '@solana/kit';
+
+try {
+    const result = await rpc.sendTransaction(wireTransaction, {
+        encoding: 'base64',
+        skipPreflight: false,
+    });
+} catch (e) {
+    if (isSolanaError(e, SOLANA_ERROR__JSON_RPC__SERVER_ERROR_SEND_TRANSACTION_PREFLIGHT_FAILURE)) {
+        // Simulation failed, but why?
+        const underlyingError = e.cause;
+        if (
+            isSolanaError(
+                underlyingError,
+                SOLANA_ERROR__INSTRUCTION_ERROR__COMPUTATIONAL_BUDGET_EXCEEDED,
+            )
+        ) {
+            // Now we know the root cause of the simulation failure, and can advise the user
+        }
+    }
+}
+```
+
+## Error codes
```

Comment:
This section feels a bit out of place because the reader is no longer a Kit consumer but a Kit contributor. I think one way to make this more explicit would be to: a) move this section at the end of the page (i.e. after "Error messages") and b) rename it something like "Contributing" so it is more obvious who we are writing for.

Alternatively, we can remove this section from the documentation altogether and add this to a `CONTRIBUTING.md` file on the package.